HW1/P5/P5_bfs.py

```diff
@@ -0,0 +1,221 @@
+import pickle
+import string
+import re
+import urllib2
+import itertools
+import math as ma
+import os
+import pyspark
+
+sc = pyspark.SparkContext()
+sc.setLogLevel('WARN')
+
+def link_string_to_KV(s):
+    src, dests = s.split(': ')
+    dests = [int(to) for to in dests.split(' ')]
+    return (int(src), dests)
+
+
+def fun(i):
+    return lambda x: (x[0], i + 1)
+
+
+def copartitioned(RDD1, RDD2):
+    "check if two RDDs are copartitioned"
+    return RDD1.partitioner == RDD2.partitioner
+
+
+def find_distance(node, dist):
+        for i in dist:
+                if i[0] == node:
+                        position = i[1]
+        return position
+
+
+links = sc.textFile('s3://Harvard-CS205/wikipedia/links-simple-sorted.txt', 32)
+page_names = sc.textFile('s3://Harvard-CS205/wikipedia/titles-sorted.txt', 32)
+
+page_names = page_names.zipWithIndex().map(lambda (n, id): (n, id + 1))
+
+Kevin_Bacon = page_names.filter(lambda (K, V): K == 'Kevin_Bacon').collect()

```

Comment:
I think this could also have been accomplished with a lookup. Not sure which is faster.